test/integration/stats/data/service_controller_test.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Stats::Data::ServicesControllerTest < ActionDispatch::IntegrationTest
+  def setup
+    @application = FactoryBot.create :cinstance
+    host! @application.provider_account.admin_domain
+    user = @application.provider_account.admins.first!
+    @token = FactoryBot.create(:access_token, owner: user, scopes: %w[stats], permission: 'rw').value
+  end
+
+  attr_reader :application, :token
+
+  test 'top applications' do
+    params = { access_token: token, since: '2021-08-30 00:00:00', period: :day, metric_name: 'hits' }
+    get top_applications_stats_data_services_path(application.service, format: :json), params: params
+    assert_response :success
+  end
+
+  test 'missing params' do
+    params = { access_token: token, period: :day, metric_name: 'hits' }
+    get top_applications_stats_data_services_path(application.service, format: :json), params: params
```

Comment:
We can dry this by extracting `params` in [the test above](https://github.com/3scale/porta/pull/2732/files#diff-c1f1a7109b968f74461fa8fa1bb44594459e5be05ad77ec1214af9f842333b4eR16)
```suggestion
    get top_applications_stats_data_services_path(application.service, format: :json), params: params.except(:period)
```